app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerEffectHandler.kt

```diff
@@ -18,6 +25,20 @@ class TextInputDatePickerEffectHandler @AssistedInject constructor(
         .subtypeEffectHandler<TextInputDatePickerEffect, TextInputDatePickerEvent>()
         .addAction(DismissSheet::class.java, uiActions::dismissSheet, schedulersProvider.ui())
         .addAction(HideDateErrorMessage::class.java, uiActions::hideErrorMessage, schedulersProvider.ui())
+        .addConsumer(ShowDateValidationError::class.java, ::applyValidationErrors, schedulersProvider.ui())
         .build()
   }
+
+  private fun applyValidationErrors(effect: ShowDateValidationError) {
+    showDateValidationErrors(effect.dateValidation)
+  }
+
+  private fun showDateValidationErrors(dateValidation: Result) {
+    when (dateValidation) {
+      is InvalidPattern -> uiActions.showInvalidDateError()
+      is DateIsInPast -> uiActions.showDateIsInPastError()
+      is MaximumAllowedDateRange -> uiActions.showMaximumDateRangeError()
+      is Valid -> throw IllegalStateException("Date validation error cannot be $dateValidation")
+    }.exhaustive()
+  }
```

Comment:
Can this be done in a single method instead?

Reply:
I don't understand. What do you mean in a single method?

Reply:
Current the only thing that's happening in `applyValidationErrors` is calling `showDateValidationErrors(effect.dateValidation)`. Can we instead just use one method?

Reply:
We haven't combined these functions yet?